design/one-pager-package-annotations.md

```diff
@@ -0,0 +1,82 @@
+
+## Background
+
+[Packages][packages-v2] extend Crossplane with new functionality. The `Provider`
+and `Configuration` Crossplane resources are powered by packages; they represent
+a declarative intent to install a provider or configuration package.
+
+Crossplane packages are OCI images that deliver a payload of valid Kubernetes
+resources. Provider packages deliver CRDs for the managed resources a provider
+defines, while Configurations deliver XRDs and Compositions. All packages
+contain general package metadata in the form of a `crossplane.yaml` file, per
+the [package format one pager][format-v2].
+
+The `crossplane.yaml` file contains Kubernetes-like YAML, and is of kind
+`Provider` or `Configuration` in API group `meta.pkg.crossplane.io`. The package
+manager propagates any annotations from the `crossplane.yaml` file to its
+corresponding `ProviderRevision` or `ConfigurationRevision` in the Kubernetes
+API server. Such annotations are referred to as 'passthrough metadata' about a
+package, because they are not meaningful to Crossplane or the package manager.
+
+The package format one pager provides an example of a `crossplane.yaml` with
+annotations such as `company` and `maintainer`, but is not opinionated about
+the format of the annotations, or which annotations should exist.
+
+## Goals
+
+The goal of this document is to encourage - but not require - consistent
+package annotations. e.g. To increase the likelihood that all package authors
+will align on using the `maintainer` annotation to indicate the package's
+maintainer.
+
+## Proposal
+
+This document recommends that packages include the following annotations:
+
+```yaml
+apiVersion: meta.pkg.crossplane.io/v1alpha1
+kind: Provider
+metadata:
+  name: example
+  annotations:
+    # The package's maintainers, as a short opaque text string.
+    meta.crossplane.io/maintainer: Example T. Maintainer <example@example.org>
+
+    # The URL at which the package's source can be found.
+    meta.crossplane.io/source: https://github.com/crossplane/provider-example
+
+    # The license under which the package's source is released.
+    meta.crossplane.io/license: Apache-2.0
+
+    # A one sentence description of the package.
+    meta.crossplane.io/description: |
+      The Google Cloud Platform (GCP) Crossplane provider adds support for
+      managing GCP resources in Kubernetes.
+
+    # A longer description, documentation, etc.
+    meta.crossplane.io/readme: |
+      provider-example is a really great Crossplane provider...
+
+    # The provider's SVG icon URI. The SVG should be optimized for 24, 48, 65px
+    # and 2x versions.
+    meta.crossplane.io/iconData: data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciLz4KBIWXMA
+
+    # A mapping of concepts to friendly, human centric names.
+    friendly-name.meta.crossplane.io/database.example.org: "Databases"
+    friendly-name.meta.crossplane.io/database.example.org/ExampleDBInstance: "ExampleDB Instance"
```

Comment:
What does this exactly refer to?

Reply:
Hmm got it. Would the format used in CRD names be more familiar to users, i.e. `friendly-name.meta.crossplane.io/exampledbinstances.database.example.org`? Though we'd probably need a differentiator between group friendly-name key and kind friendly-name key to make it easier for clients to parse.

Also, the group name is plural while this is singular. Should we go with the same choice for both?

Reply:
I really don't like having the upper-case as differentiator there since it's neglected in most of interactions of k8s users. Maybe `friendly-group-name.meta.crossplane.io` and `friendly-kind-name.meta.crossplane.io` or `kind-name...` and `group-name...`?

Reply:
I `friendly-group-name` and `friendly-kind-name` are good ideas, but we already have one consumer (Upbound Cloud) using the syntax proposed here, so my preference is to stick with this unless we think it's likely to be an issue.

FWIW despite `kubectl` etc using lowercase, I believe actual API types enforce title case kinds (i.e. they're case sensitive in the actual API). We expect these annotations to be written and consumed by fairly savvy folks so I'm don't personally see too much risk in requiring the kind case in the annotation to match the API.

Reply:
Okay, I've updated the PR to incorporate this feedback. :)